test/e2e/standalone/basic_metrics_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_standalone
+
+import (
+	"fmt"
+
+	"k8s.io/node-problem-detector/pkg/util/metrics"
+	"k8s.io/node-problem-detector/test/e2e/lib/gce"
+	"k8s.io/node-problem-detector/test/e2e/lib/npd"
+
+	"github.com/onsi/ginkgo"
+	"github.com/pborman/uuid"
+)
+
+var _ = ginkgo.Describe("NPD should export Prometheus metrics for detected problems", func() {
+	var instance gce.Instance
+
+	ginkgo.BeforeEach(func() {
```

Comment:
Actually creating instance for each test is my current plan.

Although it's slow (typically 30-180s, sometimes 600s), but we can parallelize the tests in future (I plan to do it in the next 1-2 PR). Since the metric-only tests will use only 1 instance at a time, we can easily run 20-50 of these tests in one GCE project in parallel.

I know that in k8s-e2e test, the default option is to reuse clusters. But I think that's mainly because creating a cluster is TOO slow. Plus a big cluster consumes a lot of GCP resource quota, so it's not trivial to run 10-20 of them in parallel.

I do not want different tests to reuse instance because I want to avoid pollution. Given that I plan to add stress & simulate pressure in some tests, I think having a clean environment will reduce the flakiness a lot.

I added a TODO for running tests in parallel.

Reply:
SGTM